cmd/init.go

```diff
@@ -38,180 +46,279 @@ func getInitCommand() *cobra.Command {
+				return fmt.Errorf("error finding tink-server deployment: %w", err)
 			}
 
-			log.Infof("Colony helm chart repository updated")
+			k8sClient.WaitForDeploymentReady(ctx, tinkServerDeployment, 120)
 
-			// Determine if helm release has already been installed
-			res, err = exec.ExecuteCommand(
-				log,
-				"helm",
-				"-n",
+			tinkControllerDeployment, err := k8sClient.ReturnDeploymentObject(
+				ctx,
+				"app",
+				"tink-controller",
 				"tink-system",
-				"list",
-				"-o",
-				"yaml",
-				"-A",
+				180,
 			)
 			if err != nil {
-				log.Infof("error listing current helm repositories installed: %s", err)
+				return fmt.Errorf("error finding tink-controller deployment: %w", err)
 			}
 
-			type helmRelease struct {
-				AppVersion string `yaml:"app_version"`
-				Chart      string `yaml:"chart"`
-				Name       string `yaml:"name"`
-				Namespace  string `yaml:"namespace"`
-				Revision   string `yaml:"revision"`
-				Status     string `yaml:"status"`
-				Updated    string `yaml:"updated"`
+			k8sClient.WaitForDeploymentReady(ctx, tinkControllerDeployment, 120)
+
+			k8sClient, err = k8s.New(log, filepath.Join(pwd, constants.KubeconfigHostPath))
+			if err != nil {
+				return fmt.Errorf("error creating Kubernetes client: %w", err)
 			}
 
-			var existingHelmReleases []helmRelease
-			if err := yaml.Unmarshal([]byte(res), &existingHelmReleases); err != nil {
-				return fmt.Errorf("could not get existing helm releases: %w", err)
+			log.Info("Applying tink templates")
+			err = createDirIfNotExist(filepath.Join(pwd, "templates"))
+			if err != nil {
+				return fmt.Errorf("error creating directory: %w", err)
 			}
 
-			chartInstalled := func() bool {
-				for _, release := range existingHelmReleases {
-					if release.Name == helmChartName {
-						return true
-					}
-				}
-				return false
-			}()
-
-			if !chartInstalled {
-				installFlags := []string{
-					"install",
-					"--namespace",
-					namespace,
-					helmChartName,
-					"--version",
-					helmChartVersion,
-					"konstruct/colony",
-					"--set",
-					"colony-agent.extraEnvSecrets.API_TOKEN.name=colony-api",
-					"--set",
-					"colony-agent.extraEnvSecrets.API_TOKEN.key=api-key",
-					"--set",
-					fmt.Sprintf("colony-agent.extraEnv.LOAD_BALANCER=%s", loadBalancer),
-					"--set",
-					fmt.Sprintf("colony-agent.extraEnv.COLONY_API_URL=%s", apiURL),
-					"--set",
-					fmt.Sprintf("colony-agent.extraEnv.TALOS_URL_FILES_SOURCE=http://%s:8080", loadBalancer),
-				}
+			templates := []string{"ubuntu-focal-k3s-server.yaml", "ubuntu-focal.yaml", "discovery.yaml", "reboot.yaml", "ubuntu-focal-k3s-join.yaml"}
+			for _, template := range templates {
+				url := fmt.Sprintf("https://raw.githubusercontent.com/jarededwards/k3s-datacenter/refs/heads/main/templates/%s", template)
+				filename := filepath.Join(pwd, "templates/"+template)
 
-				_, err = exec.ExecuteCommand(log, "helm", installFlags...)
+				fmt.Println(filename)
```

Comment:
This seems like quite a random `fmt.Println()`, and doesn't use the logger. Perhaps changing it to a logger?

```suggestion
				log.Infof("downloading into:", filename)
```